check with jps that the pid file contains a pid that actually is an e…

brwe opened the pull request:
…lasticsearch process

relates to #12063

rmuir commented on dev-tools/src/main/resources/ant/integration-tests.xml:
```diff
@@ -227,6 +227,27 @@
+      See if we can find the line "pid org.elasticsearch.bootstrap.Elasticsearch" in the output of jps -l.-->
+      <exec executable="jps">
+        <arg value="-l"/>
+        <redirector outputproperty="jps.pid.line">
```
for extra safety, you can declare this one as local before the exec statement (otherwise its global). In some cases maven propagates properties from ant back to itself, so in general the logic has been paranoid here to avoid something bad happening.

rmuir commented:
Looks great. This extra check is really needed.

rmuir commented on dev-tools/src/main/resources/ant/integration-tests.xml:
```diff
@@ -227,6 +227,27 @@
       <echo>Shutting down external node PID ${integ.pid}</echo>
+      <!-- verify with jps that this actually is the correct pid.
+      See if we can find the line "pid org.elasticsearch.bootstrap.Elasticsearch" in the output of jps -l.-->
+      <exec executable="jps">
```
I would maybe add failOnError=true. And i wonder if its better to set a property like this first, and use it as the executable:

```<property name=jps_executable location="${JAVA_HOME}/bin/jps"/>```

This could make it less finicky to things like build servers, as well as ensure we are using the matching jps from the jvm we used to launch ES.

rmuir replied:
i edited this to what i think is right, sorry, untested.

brwe commented:
addressed all comments. I cannot use `${java.home}/bin/jps` because `java.home` seems to point to the jre and not jdk but I can use `${environment.JAVA_HOME}/bin/jps`. Let me know if this is OK or if should go back to just calling `jps`.

s1monw commented:
LGTM

s1monw commented:
my build now fails with this:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.8:run (integ-teardown) on project elasticsearch: An Ant BuildException has occured: The following error occurred while executing this line:
[ERROR] /media/benchmark/elasticsearch/distribution/tar/target/dev-tools/ant/integration-tests.xml:299: The following error occurred while executing this line:
[ERROR] /media/benchmark/elasticsearch/distribution/tar/target/dev-tools/ant/integration-tests.xml:229: Execute failed: java.io.IOException: Cannot run program "/media/work/jdk1.7.0_55/bin/java/bin/jps" (in directory "/media/benchmark/elasticsearch/distribution/tar"): error=20, Not a directory
[ERROR] around Ant part ...<ant antfile="/media/benchmark/elasticsearch/distribution/tar/target/dev-tools/ant/integration-tests.xml" target="stop-external-cluster"/>... @ 4:141 in /media/benchmark/elasticsearch/distribution/tar/target/antrun/build-main.xml
```

I guess it's related to this, not sure if it's on my end or a configuration problem?

brwe commented:
@s1monw I assume JAVA_HOME is set to /media/work/jdk1.7.0_55/bin/java/ on your machine? Is that supposed to be so? It would work if you set it to /media/work/jdk1.7.0_55/ I guess. Can you try? Or is jps actually located at /media/work/jdk1.7.0_55/bin/java/bin/jps ?

In any way, you are already the second that reported problems with JAVA_HOME not be set to the expected path. I opened already https://github.com/elastic/elasticsearch/pull/13124 to make sure tests fail fast in case JAVA_HOME is not set at all. So, I do wonder if we should use JAVA_HOME at all. I am not sure where to get the path to jps from though.
